src/transport/tls/mod.rs

```diff
@@ -73,3 +73,7 @@ impl fmt::Display for Status {
     }
 }
 
+/// Fetch the `tls::Status` of this type.
+pub trait TlsStatus {
```

Comment:
nit/tioli: should this be called `HasStatus`? This might be less ambiguous as `tls::HasStatus` and `tls::Status`, since (I think?) we sometimes import tls::Status as TlsStatus...

not a blocker

Reply:
I'm more used to traits being "things", typically named after their methods... And many of our `Has*` traits have a method that returns an `Option<T>`, making me feel like `HasStatus` implies the status is optional... 🤷‍♂️

Reply:
although, `tls::Status` is _effectively_ an optional, since it's Conditional<T> -- not a big deal either way

Reply:
That's a good point, I had forgotten. I'll make the change.